### PR TITLE
[docs] Add landing page for GraphQL and fix small typos

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

Add landing page for GraphQL reference page and fix small typos.

## Test Plan 

image

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: docs/content/references/sui-api/graphql.mdx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: Overview
```
Comment (ronny-mysten) [inline docs/content/references/sui-api/graphql.mdx:2]:
```suggestion
title: Sui GraphQL RPC Overview
```

File: docs/content/references/sui-api/graphql.mdx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+description: GraphQL Reference page overview
```
Comment (ronny-mysten) [inline docs/content/references/sui-api/graphql.mdx:3]:
```suggestion
description: The Sui GraphQL RPC is a public service that enables interacting with the Sui network. 
```

File: docs/content/references/sui-api/graphql.mdx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+---
+
+The Sui GraphQL RPC is a public good that enables interacting with the Sui [network](https://sui.io/networkinfo).
```
Comment (ronny-mysten) [inline docs/content/references/sui-api/graphql.mdx:6]:
```suggestion
The Sui GraphQL RPC is a public service that enables interacting with the Sui [network](https://sui.io/networkinfo).
```
Can we use `service`? 'good' I think has too many opportunities to be misunderstood by esl folks.

**Review by ronny-mysten — APPROVED** [review]
I'll make a Jira to improve this page. Not that there's anything wrong with it, but I think with more time to dedicate and some customer feedback on the service, we can augment.

**Comment by amnn** [comment]
Thank you!

**Pull request merged**